src/packages/cli/build.js

```diff
@@ -1,13 +1,51 @@
 /* eslint-disable @typescript-eslint/no-var-requires */
 const { exec: nodeExec } = require('child_process');
+const path = require('node:path');
+const fs = require('node:fs/promises');
 const { promisify } = require('util');
 
 const { build } = require('esbuild');
 const { dependencies, devDependencies } = require('./package.json');
 
 const exec = promisify(nodeExec);
 
+const validateMikroOrmExpectedVersion = async () => {
+	// The CLI has the introspection operation. We need to make sure our target
+	// MikroORM version matches with the version specified in the mikro-orm adapter
+	// package.json file or the build should fail.
+	const {
+		devDependencies: mikroDevDependencies,
+		peerDependencies: mikroPeerDependencies,
+	} = require(path.resolve('..', 'mikroorm', 'package.json'));
+
+	// Let's make sure first that the version in dev and peer match.
+	const devVersion = mikroDevDependencies['@mikro-orm/core'];
+	const peerVersion = mikroPeerDependencies['@mikro-orm/core'];
+	if (devVersion !== peerVersion) {
+		throw new Error(
+			`The version of @mikro-orm/core in the mikroorm package's devDependencies (${devVersion}) and peerDependencies (${peerVersion}) do not match.`
+		);
+	}
+
+	// Ok, what's ours?
+	const searchPattern = /export const MIKRO_ORM_TARGET_VERSION = '(.+)';/;
```

Comment:
Should we move it to a config json rather than string match? My concern here is that someone breaks the string match by accident.

Reply:
Yup, I'm down, just didn't want to involve `tsx` in the build process. I'll JSON-ify it.

Reply:
Even better, I fixed it by just deriving it from the package.json file of the mikroorm package so there's no constant to update.